src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs

```diff
@@ -115,6 +116,93 @@ async Task EchoServer(ConnectionContext connection)
             }
         }
 
+#if LIBUV
+        [OSSkipCondition(OperatingSystems.Windows, SkipReason = "Libuv does not support unix domain sockets on Windows.")]
+#else
+        [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, WindowsVersions.Win8, WindowsVersions.Win81, WindowsVersions.Win2008R2, SkipReason = "UnixDomainSocketEndPoint is not supported on older versions of Windows")]
+#endif
+        [ConditionalFact]
+        [CollectDump]
+        public async Task TestUnixDomainSocketWithUrl()
+        {
+            var path = Path.GetTempFileName();
+            var url = $"http://unix:/{path}";
+
+            Delete(path);
+
+            try
+            {
+                var hostBuilder = TransportSelector.GetWebHostBuilder()
+                    .UseUrls(url)
+                    .UseKestrel()
+                    .ConfigureServices(AddTestLogging)
+                    .Configure(app =>
+                    {
+                        app.Run(async context =>
+                        {
+                            var buffer = new byte[(int)context.Request.ContentLength];
+
+                            await context.Request.Body.ReadAsync(buffer);
+                            await context.Response.WriteAsync(Encoding.ASCII.GetString(buffer));
+                        });
+                    });
+
+                using (var host = hostBuilder.Build())
+                {
+                    await host.StartAsync();
+
+                    using (var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                    {
+                        await socket.ConnectAsync(new UnixDomainSocketEndPoint(path));
```

Comment:
Why is this code different from the test above? Can you make it match exactly?

Reply:
I think, a URL which like "http://unix:/c:/temp/pipe.socket" should support HTTP, that's why I removed EchoServer. In addition to that, is there any way to implement nonhttp middleware without using listen endpoint? 

I'm sorry if I lack information about this domain.

Reply:
Ah, thats fine. It doesn't need to for this specific case.